### PR TITLE
Support single stage rotated detector in MMRotate

### PR DESCRIPTION
**Pull request by zytx121**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily receiving feedbacks. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Support single stage rotated detector in MMRotate. (#312 is included.)

- [x] add mmrotate.deploy
- [x] add rewritings for mmrotate models
- [x] add mmrotate deploy configs
- [x] add unittests
- [x] add docs & update docs
- [x] align with pytorch

<div style="margin-left: 25px;">
<table class="docutils">
<thead>
  <tr>
    <th align="center" colspan="4">MMRotate</th>
    <th align="center">Pytorch</th>
    <th align="center">ONNXRuntime</th>
    <th align="center" colspan="2">TensorRT</th>
    <th align="center">PPLNN</th>
    <th align="center">OpenVINO</th>
    <th align="left">Model Config</th>
  </tr>
</thead>
<tbody>
  <tr>
    <td align="center">Model</td>
    <td align="center">Task</td>
    <td align="center">Dataset</td>
    <td align="center">Metrics</td>
    <td align="center">fp32</td>
    <td align="center">fp32</td>
    <td align="center">fp32</td>
    <td align="center">fp16</td>
    <td align="center">fp16</td>
    <td align="center">fp32</td>
    <td>model config file</td>
  </tr>
  <tr>
    <td align="center" rowspan="2">RotatedRetinaNet</td>
    <td align="center" rowspan="2">Rotated Detection</td>
    <td align="center" rowspan="2">DOTA-v1.0</td>
    <td align="center">mAP</td>
    <td align="center">0.698</td>
    <td align="center">0.698</td>
    <td align="center">-</td>
    <td align="center">-</td>
    <td align="center">-</td>
    <td align="center">-</td>
    <td rowspan="2">$MMROTATE_DIR/configs/rotated_retinanet/rotated_retinanet_obb_r50_fpn_1x_dota_le135.py</td>
  </tr>
</tbody>
</table>
</div>

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit tests to ensure the correctness.
3. If the modification has a dependency on downstream projects of a newer version, this PR should be tested with all supported versions of downstream projects.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

**Comment by zytx121** [comment]
New PR of https://github.com/open-mmlab/mmdeploy/pull/349

**Comment by zytx121** [comment]
> I test with onnxruntime and sdk backend on windows, but it seems like there has memory leak problem. What is the memory usage when you test?

@irexyc 
The memory usage is about 1~2G in my device.

**Review by grimoire — APPROVED** [review]

**Comment by irexyc** [comment]
Can you test with onnxruntime-gpu?

I still have memory leak when device is cuda. 

My env is pytorch1.8.1+cu111, onnxruntime-gpu1.8.1

Don't know if it is a [onnxruntime bug](https://github.com/microsoft/onnxruntime/issues/8147)

File: mmdeploy/codebase/mmrotate/deploy/rotated_detection_model.py [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+
+        for i in range(batch_size):
+            dets, labels = batch_dets[i], batch_labels[i]
+            if rescale:
+                scale_factor = img_metas[i]['scale_factor']
+
+                if isinstance(scale_factor, (list, tuple, np.ndarray)):
+                    assert len(scale_factor) == 4
+                    scale_factor = np.array(scale_factor)[None, :]  # [1,4]
+                scale_factor = torch.from_numpy(scale_factor).to(
```
Comment (irexyc) [inline mmdeploy/codebase/mmrotate/deploy/rotated_detection_model.py:108]: When using onnxruntime-gpu and run test.py with --device=cuda, the scale_factor is on cuda but dets is on cpu, so `dets[:, :4] /= scale_factor` will throw an error.

**Review by irexyc — APPROVED** [review]
LGTM

**Review by grimoire — APPROVED** [review]

**Comment by lvhan028** [comment]
ci will be fixed in PR #441

**Pull request merged**